spring-cloud-sleuth-core/src/main/java/org/springframework/cloud/sleuth/instrument/web/TraceWebFilter.java

```diff
@@ -199,7 +200,18 @@ public void subscribe(CoreSubscriber<? super Void> subscriber) {
 
 			@Override
 			public void onSubscribe(Subscription subscription) {
-				this.actual.onSubscribe(subscription);
+				this.actual.onSubscribe(new Subscription() {
+					@Override
+					public void request(long n) {
+						subscription.request(n);
+					}
+
+					@Override
+					public void cancel() {
+						terminateSpan(CancelledSubscriptionException.INSTANCE);
```

Comment:
I generally don't like publicly exposing new exception types especially as it encourages people to lock to instrumentation code types. Secondarily, the only effect desired is to add an error tag and allocating exceptions isn't cheap.

If we think cancel is an error, which is debatable, please delete the exception and just add a tag("error", "canceled");

if there is an existing practice in webflux calling canceled things as exceptions, we could revisit this as well.

Reply:
I notice this is a singleton exception, which is even more odd even if allocations are fixed. :) folks will expect the stack trace to be correct and it definitely won't be. let's just use tag and skip the ceremony.

Reply:
Cool, will switch to using tag directly. If I use tag directly I can use another tag name than `error` any preference? `cancel=true`?